plugins/discovery-azure/src/main/java/org/elasticsearch/cloud/azure/management/AzureComputeServiceImpl.java

```diff
@@ -54,28 +57,43 @@ public AzureComputeServiceImpl(Settings settings) {
         KeyStoreType keystoreType = Management.KEYSTORE_TYPE_SETTING.get(settings);
 
         logger.trace("creating new Azure client for [{}], [{}]", subscriptionId, serviceName);
-        ComputeManagementClient result;
         try {
-            // Check that we have all needed properties
-            Configuration configuration = ManagementConfiguration.configure(Management.ENDPOINT_SETTING.get(settings),
-                subscriptionId, keystorePath, keystorePassword, keystoreType);
-            result = ComputeManagementService.create(configuration);
+            // Azure SDK configuration uses DefaultBuilder which uses java.util.ServiceLoader to load the
+            // various Azure services. By default, this will use the current thread's context classloader
+            // to load services. Since the current thread refers to the main application classloader it
+            // won't find any Azure service implementation.
+
+            // Here we basically create a new DefaultBuilder that uses the current class classloader to load services.
+            DefaultBuilder builder = new DefaultBuilder();
+            for (Builder.Exports exports : ServiceLoader.load(Builder.Exports.class, getClass().getClassLoader())) {
+                exports.register(builder);
+            }
+
+            // And create a new blank configuration based on the previous DefaultBuilder
+            Configuration configuration = new Configuration(builder);
+            if (logger.isTraceEnabled()) {
+                configuration.setProperty(Configuration.PROPERTY_LOG_HTTP_REQUESTS, logger.isTraceEnabled());
+            }
+
+            Configuration managementConfig = ManagementConfiguration.configure(null, configuration,
+                    Management.ENDPOINT_SETTING.get(settings), subscriptionId, keystorePath, keystorePassword, keystoreType);
+
+            logger.debug("creating new Azure client for [{}], [{}]", subscriptionId, serviceName);
+            client = ComputeManagementService.create(managementConfig);
         } catch (IOException e) {
-            logger.error("can not start azure client: {}", e.getMessage());
-            result = null;
+            throw new ElasticsearchException("Unable to configure Azure compute service", e);
         }
-        this.computeManagementClient = result;
     }
 
     @Override
     public HostedServiceGetDetailedResponse getServiceDetails() {
-        if (computeManagementClient == null) {
+        if (client == null) {
```

Comment:
I think we can now remove this condition as the client can not be null because we throw now `new ElasticsearchException("Unable to configure Azure compute service", e);` in the CTOR